### PR TITLE
py-barnaba: submission

### PR DESCRIPTION
**Pull request by GiovanniBussi**
#### Description

New port submission. Python package for analysis of RNA structures and trajectories.

**Before merging I need some advise.**

1. The barnaba package can be obtained from github (https://github.com/srnas/barnaba). It is still in a pre-release stage, so I would use the date and hashtag in order to indicate the version. Is this ok? Should I tag the version with something like 0.20180312 so that when 1.0 will be out it will be seen as an upgrade?

2. (this requires someone with python+MacPorts experience) Unfortunately, barnaba setup.py uses `setuptools_scm`, that does not allow installing from a git archive. There are two possible solutions:
a. Install from a git fetch (this is the one I implemented in the Portfile included in this pull request). In this manner, MacPorts uses `git clone` every time and `setuptools_scm` seems to work correctly. The negative side is that it does not allow caching the source and verifying the checksums.
b. Use the [setuptools_scm_git_archive](https://github.com/Changaco/setuptools_scm_git_archive) package as well. This would require modifying the setup.py upstream and add a dependency on an additional python package that, by the way, is not currently available on MacPorts.
Are there better alternatives?

Many thanks in advance!!

Giovanni

###### Tested on

macOS 10.11.6 15G19009
Xcode 8.1 8B62 

###### Verification 
Have you

- [X] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [X] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [X] checked your Portfile with `port lint`?
- [X] tried existing tests with `sudo port test`?
- [X] tried a full install with `sudo port -vst install`?
- [X] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by GiovanniBussi** [comment]
Sorry I think I found the solution by myself. Installing from pypi (where barnaba is available with a prerelease number 0.1.4) seems to solve both problems. So, in case MacPorts developers agree, I would be happy if this request merge (you can just squash merge to remove the first commit using github instead of pypi).

Thanks again!

Giovanni

**Comment by pmetzger** [comment]
We prefer that you squash the merge. Just squash and repush it.

File: python/py-barnaba/Portfile [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+master_sites        pypi:b/barnaba
+distname            barnaba-${version}
+
+name                py-barnaba
+platforms           darwin
+license             GPL-3
+maintainers         {gmail.com:giovanni.bussi @GiovanniBussi} openmaintainer
+
+description         BaRNAba is a tool for analyzing RNA three-dimensional structures and simulations.
+long_description    BaRNAba is a tool for analyzing RNA three-dimensional structures and simulations.
```
Comment (pmetzger) [inline python/py-barnaba/Portfile]: Please make your long_description more descriptive if you can.

**Comment by GiovanniBussi** [comment]
@pmetzger I improved the description, included `science` as a category, and squashed. Let me know if there is anything else I should do before merging. Thanks!

**Pull request merged**